### DISCUSSION TITLE
Newspaper Layout

### DISCUSSION THREAD

**Discussion by jaksatomovic** [py-pdf/fpdf2#1221] [Q&A] [answered]
I cant figure out how to create layout in printed newspaper style. So i need to have 3 columns but article title should be above first two columns with article text inside two column and third columns is reserved for second article. third article would again be spread across two columns etc...picture [
![](https://i.etsystatic.com/14729573/r/il/a4cd8a/3602762156/il_fullxfull.3602762156_chuh.jpg)
](url)

**Comment by Lucas-C** [comment]
Hi @jaksatomovic

As I already mentioned in a comment [on Reddit](https://www.reddit.com/r/learnpython/comments/1dy39ul/comment/lc6vvha/), please provide some minimal code when asking for programming guidance: https://stackoverflow.com/help/minimal-reproducible-example

Have you tried to use `TextColumns`, as I suggested in [this Reddit comment](https://www.reddit.com/r/Python/comments/1dwlvhq/comment/lbvik58/)?

What code have you tried to use already, and what are the precise problems that you are trying to solve?

**Comment by jaksatomovic** [comment]
Hi, Yes i tried TextColumns but it just chains text one by another i can manage to put for example TextColumns with 2 columns on 2/3 of page width with title above those two columns and then 1/3 of page width i just one column. I also tried using multicell and the result was close but i want to have fixed height for multicell can not be done as I saw an regular cell doesn't wrapp text. I haven't posted any code since actually there is none. None of my attempts is working and it is not a good reference for anything....I think Ill have to look for another lib maybe :/

**Reply by andersonhc** [reply, marked as answer]
I found it interesting and decided to create a code snippet:

https://github.com/andersonhc/fpdf2-snippets/tree/main/newspaper

You will find there all the basic concepts you need.

**Comment by gmischler** [comment]
Fpdf2 is primarily designed for "normal" documents in smallish formats like A4.
That means that most of the time a text (eg. an article) or a table will occupy the full width of the page, sometimes using multiple columns.

Newspaper layouts are very tricky. they consist of a patchwork of elements, which typically are arranged in some kind of raster, but may flow in quite unexpected ways. And since the text as written rarely fits the allocated space exactly, layout designers use tricks like changing the size of images or stretching the font of headlines to make it fit.
There's specialized layout software that allows to do this easily and interactively on screen, fitting everything into the raster as needed.

Fpdf2 is not a layout software, even if it has recently grown some basic layout features. It is very unlikely that it will ever directly support complicated tasks like newspaper layouts. For those, the client software will have to manage the distribution of text columns and images within the design grid. @andersonhc has given you an example with a few pointers about a possible approach.

Now if someone were to write a front-end package for creating newspaper layouts using fpdf2 as a backend, we would certainly all be very happy to see that. 📰 👍
